pandas/tests/frame/methods/test_join.py

```diff
@@ -104,6 +107,166 @@ def test_suffix_on_list_join():
     tm.assert_frame_equal(arr_joined, norm_joined)
 
 
+def test_join_on_single_col_check_dup():
```

Comment:
Is this test similar to the `_correctness` tests? Maybe can be combined with `@pytest.mark.parameterize`?

Reply:
Sure, that's great.